object_storage_quota.go

```diff
@@ -0,0 +1,43 @@
+package linodego
+
+import (
+	"context"
+)
+
+// ObjectStorageQuota represents a Object Storage related quota information on your account.
+type ObjectStorageQuota struct {
+	QuotaID        string `json:"quota_id"`
+	QuotaName      string `json:"quota_name"`
+	EndpointType   string `json:"endpoint_type"`
+	S3Endpoint     string `json:"s3_endpoint"`
+	Description    string `json:"description"`
+	QuotaLimit     int    `json:"quota_limit"`
+	ResourceMetric string `json:"resource_metric"`
+}
+
+// ObjectStorageQuotaUsage is the usage data for a specific Object Storage related quota on your account.
+type ObjectStorageQuotaUsage struct {
+	QuotaLimit int  `json:"quota_limit"`
+	Usage      *int `json:"usage"`
+}
+
+// ListObjectStorageQuotas lists the active ObjectStorage-related quotas applied to your account.
+// ObjectStorage Quota related features are under v4beta and may not currently be available to all users.
+func (c *Client) ListObjectStorageQuotas(ctx context.Context, opts *ListOptions) ([]ObjectStorageQuota, error) {
+	return getPaginatedResults[ObjectStorageQuota](ctx, c, formatAPIPath("object-storage/quotas"), opts)
+}
+
+// GetObjectStorageQuota gets information about a specific ObjectStorage-related quota on your account.
+// The operation includes any quota overrides in the response.
+// ObjectStorage Quota related features are under v4beta and may not currently be available to all users.
+func (c *Client) GetObjectStorageQuota(ctx context.Context, quotaID string) (*ObjectStorageQuota, error) {
+	e := formatAPIPath("object-storage/quotas/%s", quotaID)
+	return doGETRequest[ObjectStorageQuota](ctx, c, e)
+}
+
+// GetObjectStorageQuotaUsage gets usage data for a specific ObjectStorage Quota resource you can have on your account and the current usage for that resource.
```

Comment:
nitpick: Could `ObjectStorage` become `Object Storage` in the documentation to improve readability a bit?